test/ContractInteractor.test.ts

```diff
@@ -105,171 +107,138 @@ describe('ContractInteractor', () => {
-                contractInteractor.verifyForwarder(
-                    null,
-                    fakeRelayRequest,
-                    fakeSignature
-                ),
-                error.message
-            );
+    describe('getSmartWalletAddress', function () {
+        let smartWalletFactory: ISmartWalletFactoryInstance;
+        const owner = '0x2';
+        const recoverer = constants.ZERO_ADDRESS;
+        const index = '1';
+        const smartWalletAddress = '0x1';
+
+        beforeEach(function () {
+            smartWalletFactory = stubInterface<ISmartWalletFactoryInstance>({
+                getSmartWalletAddress: Promise.resolve(smartWalletAddress)
+            });
         });
 
-        it('should fail if RelayRequest is null', async () => {
-            const error = new TypeError(
-                "Cannot read properties of null (reading 'relayData')"
-            );
-            fakeIForwarderInstance.verify.throwsException(error);
-            await assert.isRejected(
-                contractInteractor.verifyForwarder(
-                    fakeSuffixData,
-                    null,
-                    fakeSignature
-                ),
-                error.message
-            );
+        afterEach(function () {
+            sinon.restore();
         });
 
-        it('should fail if Signature is null', async () => {
-            const error = new TypeError(
-                "Cannot read properties of null (reading 'length')"
-            );
-            fakeIForwarderInstance.verify.throwsException(error);
-            await assert.isRejected(
-                contractInteractor.verifyForwarder(
-                    fakeSuffixData,
-                    fakeRelayRequest,
-                    null
-                ),
-                error.message
+        it('should return smart wallet address', async function () {
+            sinon
+                .stub(contractInteractor, '_createSmartWalletFactory')
+                .withArgs(address)
+                .returns(Promise.resolve(smartWalletFactory));
+            const expectedSmartWalletAddress =
+                await contractInteractor.getSmartWalletAddress(
+                    address,
+                    owner,
+                    recoverer,
+                    index
+                );
+            expect(expectedSmartWalletAddress).to.be.equal(
+                smartWalletAddress,
+                `${expectedSmartWalletAddress} should equal ${smartWalletAddress}`
             );
         });
 
-        it('should fail if callForwarder is null', async () => {
-            _createForwarderStub.restore();
-            fakeRelayRequest.relayData.callForwarder = null;
-            await assert.isRejected(
-                contractInteractor.verifyForwarder(
-                    fakeSuffixData,
-                    fakeRelayRequest,
-                    fakeSignature
-                ),
-                'Invalid address passed to IForwarder.at(): null'
```

Comment:
Why did we remove the verifyForwarder tests?

Reply:
I removed since we had a meeting where we talked about no creating unit testing for typing validation.